### PR TITLE
feat: convert to async

### LINKED ISSUE #44: Swap to async
I can probably take a look at it since I had to do it for [alexa_media_player](https://github.com/custom-components/alexa_media_player).

### PR DESCRIPTION
**Pull request by alandtse**
This version contains numerous changes in addition to the async conversion. It includes a new charging sensor, support for oauth, performance tweaks, and incorporation of black for formatting.
BREAKING CHANGE: API calls now require async
Closes #44, #47

### CODE REVIEW

**Comment by alandtse** [comment]
To test this, use this [branch](https://github.com/alandtse/home-assistant/tree/tesla_async) of HA. I will be submitting it after a few days testing.

File: Pipfile [hunk 1]
```diff
@@ -12,6 +12,10 @@ pytest-cov = "*"
 tox = "*"
 twine = "*"
 python-semantic-release = "*"
+black = "*"
 
 [packages]
-requests = "*"
+aiohttp = "*"
+
+[pipenv]
```
Comment (zabuldon) [inline Pipfile]: Let's freeze versions.

File: pylintrc [hunk 2]
```diff
@@ -1,10 +1,15 @@
+[MASTER]
+ignore=tests
```
Comment (zabuldon) [inline pylintrc]: Do we really want to ignore test in linting? In my opinion we have to keep them readable too.

File: setup.cfg [hunk 3]
```diff
@@ -3,4 +3,56 @@ version_variable=teslajsonpy/__version__.py:__version__
...
+hvcs=github
+
+[pydocstyle]
+ignore = D202, D212, D416, D213, D203, D407
+
+[flake8]
+exclude = .venv,.git,.tox,docs,venv,bin,lib,deps,build
+# To work with Black
+max-line-length = 88
+# E501: line too long
```
Comment (zabuldon) [inline setup.cfg]: Let's increase max-line-length to 120 instead of ignoring E501

Comment (alandtse) [reply setup.cfg]: I'm just taking the HA ignores for now. They default to 88 since they're using black.

Comment (zabuldon) [reply setup.cfg]: We don't need to keep HA code-style in our repo as long as it separated library.

Comment (alandtse) [reply setup.cfg]: True, but remember one of the goals #24 was to potentially move some of the HA-specific code over to HA? It'd make it easier if we enforced similar code-styles.

I'm pretty sure black would ignore the 120 length anyway and bring it back down to 88.

Comment (zabuldon) [reply setup.cfg]: @alandtse ok.

File: pylintrc (hunk 2 above)
Comment (alandtse) [reply pylintrc]: Ok, removing the ignore.

File: Pipfile (hunk 1 above)
Comment (alandtse) [reply Pipfile]: Can't we rely on the Pipfile.lock file?

**Comment by alandtse** [comment]
Whenever you're ready to merge this the semantic-release workflow should kick in. It's currently set with a `--noop` flag so we can confirm it thinks the next release is v1.0.0.  We'll need to remove that --noop flag to test the github upload and pypi release. I believe all those credentials should work as the debug is showing they aren't blank (they're masked though).

**Pull request merged**